src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/FieldBuilderImpl.cs

```diff
@@ -20,13 +23,46 @@ internal FieldBuilderImpl(TypeBuilderImpl typeBuilder, string fieldName, Type ty
             _attributes = attributes & ~FieldAttributes.ReservedMask;
         }
 
-        #region MemberInfo Overrides
         protected override void SetConstantCore(object? defaultValue) => throw new NotImplementedException();
-        protected override void SetCustomAttributeCore(ConstructorInfo con, byte[] binaryAttribute) => throw new NotImplementedException();
+        protected override void SetCustomAttributeCore(ConstructorInfo con, byte[] binaryAttribute)
+        {
+            if (!IsPseudoAttribute(con.ReflectedType!.FullName!))
+            {
+                _customAttributes.Add(new CustomAttributeWrapper(con, binaryAttribute));
+            }
+        }
+
+        private bool IsPseudoAttribute(string attributeName)
+        {
+            switch (attributeName)
+            {
+                case "System.Runtime.InteropServices.FieldOffsetAttribute":
+                    /* TODO:  Not sure how to apply this
```

Comment:
This attribute is exposed via `FieldInfo.Attributes` so it seems like it needs to be added to the list.

Reply:
Not sure I understood, but there is no offset related value in [FieldAttributes](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Reflection/FieldAttributes.cs) enum.

Reply:
The offsets are not in the enum, but it should be returned from the FieldInfo.Attributes  as as a FieldOffsetAttribute

Reply:
(based on offline discussion, also see https://learn.microsoft.com/dotnet/api/system.reflection.metadata.ecma335.metadatabuilder.addfieldlayout)